diff --git a/dist/providers/anthropic.js b/dist/providers/anthropic.js
index 6abaaa94b90c31fb3c5fa87c3b9634926b4c4887..cb30a02d4aa6cebbcd1fcabe1c85c01aaec41a92 100644
--- a/dist/providers/anthropic.js
+++ b/dist/providers/anthropic.js
@@ -408,6 +408,7 @@ function createClient(model, apiKey, interleavedThinking, optionsHeaders, dynami
             betaFeatures.push("interleaved-thinking-2025-05-14");
         }
         const client = new Anthropic({
+            fetch: model.fetch,
             apiKey: null,
             authToken: apiKey,
             baseURL: model.baseUrl,
@@ -427,6 +428,7 @@ function createClient(model, apiKey, interleavedThinking, optionsHeaders, dynami
     // OAuth: Bearer auth, Claude Code identity headers
     if (isOAuthToken(apiKey)) {
         const client = new Anthropic({
+            fetch: model.fetch,
             apiKey: null,
             authToken: apiKey,
             baseURL: model.baseUrl,
@@ -443,6 +445,7 @@ function createClient(model, apiKey, interleavedThinking, optionsHeaders, dynami
     }
     // API key auth
     const client = new Anthropic({
+        fetch: model.fetch,
         apiKey,
         baseURL: model.baseUrl,
         dangerouslyAllowBrowser: true,
diff --git a/dist/providers/openai-completions.js b/dist/providers/openai-completions.js
index c37f67f164c8880474cfd2c221d130594c273d08..eaa11a573af974977f82c5451ef06bfc28e213ae 100644
--- a/dist/providers/openai-completions.js
+++ b/dist/providers/openai-completions.js
@@ -286,6 +286,7 @@ function createClient(model, context, apiKey, optionsHeaders) {
         Object.assign(headers, optionsHeaders);
     }
     return new OpenAI({
+        fetch: model.fetch,
         apiKey,
         baseURL: model.baseUrl,
         dangerouslyAllowBrowser: true,
diff --git a/dist/providers/openai-responses.js b/dist/providers/openai-responses.js
index 3bf172acee9286eb71fb40c067b303ea96563486..b1b8f8779a671c940f0c2fae27e745e9d6986259 100644
--- a/dist/providers/openai-responses.js
+++ b/dist/providers/openai-responses.js
@@ -124,6 +124,7 @@ function createClient(model, context, apiKey, optionsHeaders) {
         Object.assign(headers, optionsHeaders);
     }
     return new OpenAI({
+        fetch: model.fetch,
         apiKey,
         baseURL: model.baseUrl,
         dangerouslyAllowBrowser: true,
